perfkitbenchmarker/linux_packages/cassandra.py

```diff
@@ -90,6 +94,27 @@ def AptInstall(vm):
   _Install(vm)
 
 
+def JujuInstall(vm):
+  if vm.isController is True:
+    if vm.vm_group_specs and 'workers' in vm.vm_group_specs:
+      vm.JujuDeploy('cs:trusty/cassandra',
+                    vm.vm_group_specs['workers'].vm_count,
+                    'workers')
```

Comment:
There's a certain part of the Juju installation that has to be tied to the benchmark, i.e., the cassandra charm needs to be installed into the workers group, and the cassandra-stress charm into the clients group. I'm not sure abstracting that out and still being able to maintain that benchmark-specific installation, unless I'm totally missing your point which is entirely possible.